### DISCUSSION TITLE
[BP#1] Eliminer les fonctionnalités non essentielles

### DISCUSSION THREAD

**Discussion by ddebarbieux** [cnumr/best-practices#113] [Bonnes pratiques] [answered]
Couper cette bonne pratique en deux :
-	Etape mesure : expliquer la différence en mesurer l’audience d’une page et mesurer l’usage d’une fonctionnalité.  Toutes optimisations, même celles d’éliminer une fonctionnalité, commencent par une mesure factuelle et convaincante. Cette mesure permet aussi d’orienter d’autres choix stratégique.
-	Etape de conception : reprendre l’exemple google/yahoo qui est un choix de design. Donner un autre exemple : une page administrateur peut être ‘dégelasse’.  Elle est indispensable mais utilisée 3 fois par an par un interne.

NB : Sans atteindre le niveau de yahoo, le page d’accueil de google est de plus en plus chargée 😊

**Comment by ACTLEM** [comment]
L'idée me plaît car elle permet d'insister sur cette recommandation majeure mais l'étape mesure est plus de l'outillage (il faut je pense que l'on crée vraiment une section dédiée à celà et hors bonnes pratiques ;-) ), donc j'ai du mal à voir un découpage sur deux recommandations d'écoconception avec chacune une règle de validation.

**Reply by ddebarbieux** [reply]
La formulation actuelle contient les deux notions (_Si l'application est déjà developpée..._). C'est pourquoi, je propose de la découper.

Comme le dit ACTLEM, il y a peu de bonnes pratiques fonctionnelles (comparé aux nombres de bonnes pratiques techniques). C’est un moyen de rééquilibrer.

**Comment by florinesueur** [comment, marked as answer]
Si j'ai bien compris le découpage, je propose :
- À l'étape de conception : Vérifier l'utilité de la fonctionnalité en amont de son développement par des ateliers, wireframes et prototype avec tests utilisateurs. La règle de validation pourrait être si des wireframes ont été testés ou encore 70% des fonctionnalités ont été testés avant développement.
- À l'étape de production : Mesurer l'utilisation de la fonctionnalité. La règle de validation reste la même moins de 10% de fonctionnalité peut utilisées présentent en production.

**Reply by ACTLEM** [reply]
Il y aurait donc: "Eviter de développer des fonctionnalités non essentielles" et "Eliminer les fonctionnalités non utiles" (ou du moins un truc de ce genre) avec les règles de validation proposées par @florinesueur.
@LGnord cela rejoint ta pensée?

**Reply by ddebarbieux** [reply]
Oui, pour la deuxième je dirai même : Eliminer les fonctionnalités non utilisées.

**Reply by ACTLEM** [reply]
J'ai créé les 2 issues:
- Modification BP1: https://github.com/cnumr/best-practices/issues/288
- Ajout BP "Eliminer les fonctionnalités non utilisées": https://github.com/cnumr/best-practices/issues/289

**Comment by Christophe-Cl** [comment]
Lien avec la [#107](https://github.com/cnumr/best-practices/discussions/107) ?
